v22.1/security-reference/scram-authentication.md

```diff
@@ -0,0 +1,323 @@
+
+That environment must be a provisioned with:
+
+- A tool for [creating PostgreSQL/CockroachDB-formated SCRAM-SHA-256 hashes](#managing-users-with-pre-hashed-passwords), such as the example script provided in the [appendix](#appendix-python-scram-hashing-script).
+- A CRDB client with ['USER CREATE/ALTER privileges'](../create-user.html#create-a-user-that-can-create-other-users-and-manage-authentication-methods-for-the-new-users)
+- Access, with write privileges, to a secrets store where the plaintext passwords will be persisted. Recommended solutions include Hashicorp Vault, GCP secrets manager or AWS secrets manager.
+
+On that instance, [generate SCRAM salted password hashes](#managing-users-with-pre-hashed-passwords), manage the CRDB users via CRDB client and persist the secrets in the secrets store, as described in the following section.
+
+## Managing users with pre-hashed passwords
+
+[Many database tools](https://passlib.readthedocs.io/en/stable/lib/passlib.hash.scram.html) support SCRAM-SHA-256 and can be used to calculate the hashes servers will need for authentication.
+
+In this example, we'll use this [Python script](#appendix-python-scram-hashing-script), which, like many good things, comes from the [open-source community](https://stackoverflow.com/questions/68400120/how-to-generate-scram-sha-256-to-create-postgres-13-user).
+
+### Start a python shell
+
+For example, with docker.
+
+{% include_cached copy-clipboard.html %}
+```bash
+docker run -it python:latest python
+```
+
+### Compute salted password hashes
+
+In the Python shell, use another tool or the [script in the Appendix](#appendix-python-scram-hashing-script) to create a SCRAM-SCHA-256 salted password hash in the format accepted by CockroachDB (which follows PostgreSQL's format).
+
+{% include_cached copy-clipboard.html %}
+```python
+# Either copy-paste the hashing script directly into the shell, or into a file named roach_scram_hasher.py and import by un-commenting the following line
+# from roach_scram_hasher import *
+
+pg_scram_sha256("password")
+```
+```txt
+'SCRAM-SHA-256$119680:D1nWfUQfRP2HWbVuK19Ntw==$ulRLI4fp/bvJN/oBC3x6Q8daLnPit2eWLvm2sex9vyc=:XHVeRmT8iboSn0KMbIx3nMRrebR2gIbi+qEMnHfJi+I='
+```
+
+### Create a SCRAM-SHA-256-authenticated SQL user
+
+In the CockroachDB SQL shell, use the computed hash as the value of `PASSWORD` when [creating a user](../create-user.html) or [rotating the passwors for an existing user](../alter-user.html).
+
+{% include_cached copy-clipboard.html %}
+```sql
+CREATE USER cool_user WITH PASSWORD 'SCRAM-SHA-256$119680:D1nWfUQfRP2HWbVuK19Ntw==$ulRLI4fp/bvJN/oBC3x6Q8daLnPit2eWLvm2sex9vyc=:XHVeRmT8iboSn0KMbIx3nMRrebR2gIbi+qEMnHfJi+I=';
+
+ALTER USER other_user WITH PASSWORD "$scram$100000$vZcyRiil1BojRGhN$sha-1=pRLECKY8rLwZKkmhKu02b8RfIeQ,sha-256=ClkmygvxmE9tOx8swe2lD5kb/4rjE3qa2BF6mAhshHE,sha-512=l3zLEbJLHqRZGi5uCdwDqkNzk4bfoZTcvyKRXyu6YP3mtoh7JFclxKdAKXKQTBAWMQDv.pbNgUt3/KOX4Hgteg";
```

Comment:
I might block alter user into it's own `sql` code block for clarity and its own prompt